engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -484,6 +484,30 @@ describe('System test: QuerySparql', () => {
           });
         });
       });
+      describe('functionArgumentsCache', () => {
```

Comment:
Could you also add a test to check if the default cache object (if the user doesn't provide on in the context) is reused across query execs?